shellclear/src/engine.rs

```diff
@@ -94,17 +74,25 @@ impl PatternsEngine {
     ///
     /// Will return `Err` when has an error when find sensitive patterns in a
     /// specific shell
-    pub fn find_history_commands_from_shall_list(
+    pub fn find_history_commands_from_shell_list(
         &self,
         shells_context: &Vec<ShellContext>,
-        clear: bool,
-    ) -> Result<Findings> {
-        let mut findings = Findings::default();
+    ) -> Result<(Vec<FindingSensitiveCommands>, Vec<FindingSensitiveCommands>)> {
```

Comment:
In previous suggestion i think we need to change this result to 
```rs
pub struct Findings {
    engine_kind: HashMap<Shell, Vec<&FindingSensitiveCommands>>
}
```

then we can return only `Findings` and add helper functions under this strut. 

what do you think?